lib/OpenQA/WebAPI/Controller/API/V1/Table.pm

```diff
@@ -208,6 +231,20 @@ sub update {
     my $ret;
     my $update = sub {
         my $rc = $schema->resultset($table)->find({id => $self->param('id')});
+        # Tables used in a group configured in YAML must not be renamed
+        if (
+            (($table eq 'TestSuites' || $table eq 'Machines') && $rc->name ne $self->param('name'))
+            || (
```

Comment:
this expression looks like it's better split for easier readability (hurry for local variables :)

Reply:
And `$self->param('name')` might not even be defined here as well as the other params. The entire controller actually lacks validation of input parameters...

Reply:
I'm not sure what you mean about validation here... name is required. You can't update the table without setting it. So here I'm checking if the name matches - same name means no change, different name means rename,

Reply:
I mean `$self->param('name')` might not even be defined. And `_prepare_settings` uses the `$table`/`$self->param("table")` which might not be defined as well. But yes, if the table is 'TestSuites' or 'Machines' the name is actually validated. So I guess this is good for now but we should improve the general validation at some point.